cmd/system-probe/modules/ping.go

```diff
@@ -0,0 +1,138 @@
+	"google.golang.org/grpc"
+)
+
+const (
+	countParam    = "count"
+	intervalParam = "interval"
+	timeoutParam  = "timeout"
+)
+
+type pinger struct{}
+
+// Pinger is a factory for NDMs Ping module
+var Pinger = module.Factory{
+	Name:             config.PingModule,
+	ConfigNamespaces: []string{},
+	Fn: func(cfg *sysconfigtypes.Config) (module.Module, error) {
+		return &pinger{}, nil
+	},
+	NeedsEBPF: func() bool {
+		return false
+	},
+}
+
+var _ module.Module = &pinger{}
+
+func (p *pinger) GetStats() map[string]interface{} {
+	return nil
+}
+
+func (p *pinger) Register(httpMux *module.Router) error {
+	var runCounter = atomic.NewUint64(0)
+
+	httpMux.HandleFunc("/ping/{host}", utils.WithConcurrencyLimit(utils.DefaultMaxConcurrentRequests, func(w http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		vars := mux.Vars(req)
+		id := getClientID(req)
+		host := vars["host"]
+
+		count, err := getIntParam(countParam, req)
+		if err != nil {
+			log.Errorf("unable to run ping invalid count %s: %s", host, err)
+			w.Write([]byte("invalid count"))
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		interval, err := getIntParam(intervalParam, req)
+		if err != nil {
+			log.Errorf("unable to run ping invalid interval %s: %s", host, err)
+			w.Write([]byte("invalid interval"))
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		timeout, err := getIntParam(timeoutParam, req)
+		if err != nil {
+			log.Errorf("unable to run ping invalid timeout %s: %s", host, err)
+			w.Write([]byte("invalid timeout"))
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		cfg := pingcheck.Config{
+			UseRawSocket: true,
+			Interval:     time.Duration(interval),
+			Timeout:      time.Duration(timeout),
+			Count:        count,
+		}
+
+		// Run ping using raw socket
+		result, err := pingcheck.RunPing(&cfg, host)
+		if err != nil {
+			log.Errorf("unable to run ping for host %s: %s", host, err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		resp, err := json.Marshal(result)
+		if err != nil {
+			log.Errorf("unable to marshall ping stats: %s", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		_, err = w.Write(resp)
+		if err != nil {
+			log.Errorf("unable to write ping response: %s", err)
+		}
+
+		runCount := runCounter.Inc()
+		logPingRequests(host, id, count, interval, timeout, runCount, start)
+	}))
+
+	return nil
+}
+
+func (p *pinger) RegisterGRPC(_ grpc.ServiceRegistrar) error {
+	return nil
+}
+
+func (p *pinger) Close() {}
+
+func logPingRequests(host string, client string, count int, interval int, timeout int, runCount uint64, start time.Time) {
+	args := []interface{}{host, client, count, interval, timeout, runCount, time.Since(start)}
+	msg := "Got request on /ping/%s?client_id=%s&count=%d&interval=%d&timeout=%d (count: %d): retrieved ping in %s"
+	switch {
+	case count <= 5, count%20 == 0:
+		log.Infof(msg, args...)
+	default:
+		log.Debugf(msg, args...)
+	}
```

Comment:
```suggestion
	log.Debugf(msg, args...)
```
Hum, do we need `Info` log level? Since running ping is a normal/expected operation, I think it might be better to only log as debug. WDYT?

Reply:
I copied the code directly from another module. It's not necessary I agree, but one thing to note is that this logs to `system-probe.log` which doesn't have a lot coming out of it anyway with only `ping` enabled. 

I personally like the idea of being able to see that pings are happening by default and some indicator that the integration is running in the logs. Any user doing troubleshooting can immediately see that the module is doing something without having to update a config file and restart the system.

If I keep it in, WDYT about changing the modulus to something higher? 50? 100?